### PR TITLE
[Dark Mode] Map color segmentation for showing full gray scale map on background but keeping the colored layers

### PR DESCRIPTION
**Pull request by EMREOYUN**
Inspired by the simrail.app map. However, the major change is speed and infrastructure layers keep their colors. This code only works on dark mode.

Before:
opera_1Mo1WDksVi

How it looks on simrail.app:
image

After:
opera_FC4aKhu4TS

### CODE REVIEW

**Comment by GNimrodG** [comment]
Hi,
It was intentional design on my part to use the current color scheme and not just a grayscale map.
It's a good idea, but I would like to keep the current design too, so maybe we can implement it as a setting? So users can chose which style they prefer.

**Comment by EMREOYUN** [comment]
I managed to add something like this because I couldn't manage how to reload CSS classes dynamically:
image
If that is fine, I can commit these changes.

**Comment by GNimrodG** [comment]
I think should be possible without the reload, just create an other class and toggle them based on the setting. I will look into it.

**Review by GNimrodG — CHANGES_REQUESTED** [review]
I suggested some changes that should fix the reload issue.
If you don't want to work with it anymore I will have time to fix these today after 6PM CEST.

**Comment by GNimrodG** [comment]
Also you don't need to commit each file separately, you can include them in one commit ;)

File: smo-frontend/src/components/Settings.tsx [hunk 1]
```diff
@@ -145,6 +155,33 @@ const Settings: FunctionComponent = () => {
...
+            <Typography
+              sx={{ pt: 2 }}
+              level="h4">
+              Map Theme(For Dark Mode)
+            </Typography>
+            <Typography
+                sx={{ pt: 0 }}
+                level="body-lg">
+                Reload Reqiured
+              </Typography>
```
[the remark this answers is not in the document]
Comment (GNimrodG) [reply smo-frontend/src/components/Settings.tsx:171]: Also pay attention to the formatting please, the closing tag should be at the same level as the opening tag

**Comment by EMREOYUN** [comment]
> Also you don't need to commit each file separately, you can include them in one commit ;)

Since the other branch of mine has pending commits(personal changes), I do change files in Github Web Editor which does edit one file at the time.

**Comment by GNimrodG** [comment]
> Since the other branch of mine has pending commits(personal changes), I do change files in Github Web Editor which does edit one file at the time.

You can use GitHub's web based VSCode (press `.` on the keyboard while the branch is open) then, there you can make and commit your changes.

File: smo-frontend/src/components/MainMap.tsx [hunk 2]
```diff
@@ -37,9 +37,13 @@ const MAIN_ATTRIBUTIONS = [
   'This website is not affiliated with the <a href="https://simrail.eu" target="_blank">SimRail</a> team.',
 ].join(" | ");
 
+
 const MainMap: FunctionComponent = () => {
   const { setSelectedTrain } = useContext(SelectedTrainContext);
   const { selectedRoute, setSelectedRoute } = useContext(SelectedRouteContext);
+  const [alternativeTheme] = useState(
+    readLocalStorageValue({ key: "alternativeTheme", defaultValue: false })
+  );
```
[the remark this answers is not in the document]
Comment (EMREOYUN) [reply smo-frontend/src/components/MainMap.tsx:46]: The setting is updated, which is good. However, the main map does not refresh itself dynamically as of now.

**Comment by EMREOYUN** [comment]
I wrote a custom TileLayer to refresh layers dynamically. Simple but effective.

**Review by GNimrodG — APPROVED** [review]
LGTM

**Pull request merged**